File: solutions/csharp/leap/1/Leap.cs
using System;

public static class Leap
{
    public static bool IsLeapYear(int year)
    {
            if (year % 4 == 0)
            {
                if (year % 100 == 0)
                {
                    if (year % 400 == 0)
                    {
                        return true; // Divisible by 400
                    }
                    return false; // Divisible by 100 but not 400
                }
                return true; // Divisible by 4 but not 100
            }
            return false; // Not divisible by 4
        }
    
}

File: solutions/clojure/leap/2/src/leap.clj
(ns leap)

(defn- divides? [number divisor]
  (zero? (mod number divisor)))

(defn leap-year?
  [year]
    (cond
      (divides? year 400) true
      (divides? year 100) false
      (divides? year 4) true
      :else false))
